app/scripts/migrations/135.test.ts

```diff
@@ -0,0 +1,131 @@
+import { SmartTransaction } from '@metamask/smart-transactions-controller/dist/types';
+import { migrate, VersionedData } from './135';
+
+const prevVersion = 134;
+
+describe('migration #135', () => {
+  const mockSmartTransaction: SmartTransaction = {
+    uuid: 'test-uuid',
+  };
+
+  it('should update the version metadata', async () => {
+    const oldStorage: VersionedData = {
+      meta: { version: prevVersion },
+      data: {},
+    };
+
+    const newStorage = await migrate(oldStorage);
+    expect(newStorage.meta).toStrictEqual({ version: 135 });
+  });
+
+  it('should set stx opt-in to true when stx opt-in status is null', async () => {
+    const oldStorage: VersionedData = {
+      meta: { version: prevVersion },
+      data: {
+        PreferencesController: {
+          smartTransactionsOptInStatus: null,
+        },
+      },
+    };
+
+    const newStorage = await migrate(oldStorage);
+    expect(
+      newStorage.data.PreferencesController?.smartTransactionsOptInStatus,
```

Comment:
The `PreferencesController` will always be there in this case, so we don't need `?`. It can be: `newStorage.data.PreferencesController.smartTransactionsOptInStatus`. The same in other tests.

Reply:
According to their type def:
```ts
export type VersionedData = {
  meta: {
    version: number;
  };
  data: {
    PreferencesController?: {  // Note the ? here
      preferences?: {
        smartTransactionsOptInStatus?: boolean | null;
      };
      smartTransactionsOptInStatus?: boolean | null;
    };
  };
};
```

TypeScript is correctly warning us that PreferencesController might be undefined according to the type definition. (If I don't use it I get a warning)
```
'newStorage.data.PreferencesController' is possibly 'undefined'.ts(18048)
```

Therefore, I do believe the safest thing to do is to keep the optional chaining. LMKWYT...